onboarding-enabler-java/src/main/java/org/zowe/apiml/eurekaservice/client/util/EurekaInstanceConfigCreator.java

```diff
@@ -0,0 +1,141 @@
+
+    private final MapUtils mapUtils;
+
+    public EurekaInstanceConfigCreator(MapUtils mapUtils) {
+        this.mapUtils = mapUtils;
+    }
+
+    public EurekaInstanceConfig createEurekaInstanceConfig(ApiMediationServiceConfig config) throws ServiceDefinitionException {
+        ApimlEurekaInstanceConfig result = new ApimlEurekaInstanceConfig();
+
+        String hostname;
+        int port;
+        URL baseUrl;
+
+        try {
+            baseUrl = new URL(config.getBaseUrl());
+            hostname = baseUrl.getHost();
+            port = baseUrl.getPort();
+        } catch (MalformedURLException e) {
+            String message = String.format("baseUrl: [%s] is not valid URL", config.getBaseUrl());
+            throw new ServiceDefinitionException(message, e);
+        }
+
+        result.setInstanceId(String.format("%s:%s:%s", hostname, config.getServiceId(), port));
+        result.setAppname(config.getServiceId());
+        result.setAppGroupName(config.getServiceId());
+        result.setHostName(hostname);
+        result.setIpAddress(config.getServiceIpAddress());
+        result.setInstanceEnabledOnit(true);
+        result.setSecureVirtualHostName(config.getServiceId());
+        result.setVirtualHostName(config.getServiceId());
+        result.setStatusPageUrl(config.getBaseUrl() + config.getStatusPageRelativeUrl());
+
+        if ((config.getHomePageRelativeUrl() != null) && !config.getHomePageRelativeUrl().isEmpty()) {
+            result.setHomePageUrl(config.getBaseUrl() + config.getHomePageRelativeUrl());
+        }
+
+        String protocol = baseUrl.getProtocol();
+        result.setNonSecurePort(port);
+        result.setSecurePort(port);
+
+        switch (protocol) {
+            case "http":
+                result.setNonSecurePortEnabled(true);
+                result.setHealthCheckUrl(config.getBaseUrl() + config.getHealthCheckRelativeUrl());
+                break;
+            case "https":
+                result.setSecurePortEnabled(true);
+                result.setSecureHealthCheckUrl(config.getBaseUrl() + config.getHealthCheckRelativeUrl());
+                break;
+            default:
+                throw new ServiceDefinitionException(String.format("'%s' is not valid protocol for baseUrl property", protocol));
+        }
+
+        try {
+            result.setMetadataMap(createMetadata(config));
+        } catch (MetadataValidationException | IllegalArgumentException e) {
+            throw new ServiceDefinitionException("Service configuration failed to create service metadata: ", e);
+        }
+
+        return result;
+    }
+
+    private Map<String, String> createMetadata(ApiMediationServiceConfig config) {
+        Map<String, String> metadata = new HashMap<>();
+
+        // fill authentication metadata
+        Authentication authentication = config.getAuthentication();
+        if (authentication != null) {
+            metadata.put(AUTHENTICATION_SCHEME, authentication.getScheme());
```

Comment:
empty (null) values should not be set

Reply:
I'm glad you say... I will stop by to discuss this

Reply:
as we debated yesterday, i'll leave this part be as is